### ISSUE TITLE
Keep asking for platform at every reload

### ISSUE THREAD

**Issue by shrinidhi111** [microsoft/vscode-remote-release#2934]
Version Information :
```
Version: 1.44.1 (user setup)
Commit: a9f8623ec050e5f0b44cc8ce8204a1455884749f
Date: 2020-04-11T01:48:12.622Z
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363
```

I have been using Remote SSH from a long time now and in the latest release, for the FIRST TIME EVER, it keeps asking for the remote platform.

Keeps asking each and every time.

Even after I select the remote AWS Ubuntu instance as "Linux" it won't save it. And at every Window Reload it keeps asking.

![Screenshot (11)_LI](https://user-images.githubusercontent.com/10372299/81036228-eded4580-8ebb-11ea-9f8f-a6940f0232dc.jpg)

Again and again and again.

**Comment by shrinidhi111** [comment]
Updated to latest version. Same issue

```
Version: 1.44.2 (user setup)
Commit: ff915844119ce9485abfe8aa9076ec76b5300ddd
Date: 2020-04-16T16:36:23.138Z
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363
```

**Comment by RVoigtHearsay** [comment]
The poor UX for this makes it especially maddening, with multiple window refreshes, modal dialogs and interruptive popups *every* time. It's especially irksome when working on a VPN that occasionally drops.

I tried putting this in settings.json for the remote workspace but to no avail, I still get pestered to remind VSCode that it's a linux box every reload:
```
{
    "remote.SSH.remotePlatform": "linux"
}
```

**Comment by Chuxel** [comment]
//cc: @roblourens

**Comment by roblourens** [comment]
Sorry, it's supposed to write this to your settings, I don't know why it isn't working. The correct format is like 

```
"remote.SSH.remotePlatform": {
        "my-hostname": "linux"
}
```

Can you share the output of the Remote-SSH output channel to check for errors after you select the option? Also, are there errors in the devtools (Help>Toggle Developer Tools)

**Comment by RVoigtHearsay** [comment]
Found this error in the devtools:
```
Error Python Extension: 2020-06-09 17:58:32: Failed to check if file needs to be fixed [EntryNotFound (FileSystemError): Unable to read file 'vscode-remote://ssh-remote+devbox/home/rvoigt/.config/Code/User/settings.json' (EntryNotFound (FileSystemError): Error: ENOENT: no such file or directory, open '/home/rvoigt/.config/Code/User/settings.json')	at _handleError (/home/rvoigt/.vscode-server/bin/5763d909d5f12fe19f215cbfdd29a91c0fa9208a/out/vs/server/remoteExtensionHostProcess.js:674:800)	at processTicksAndRejections (internal/process/task_queues.js:89:5)	at async v.readText (/home/rvoigt/.vscode-server/extensions/ms-python.python-2020.5.80290/out/client/extension.js:1:532085)	at async doesFileNeedToBeFixed (/home/rvoigt/.vscode-server/extensions/ms-python.python-2020.5.80290/out/client/extension.js:92:842702)	at async /home/rvoigt/.vscode-server/extensions/ms-python.python-2020.5.80290/out/client/extension.js:92:841704	at async Promise.all (index 0)	at async d.getFilesToBeFixed (/home/rvoigt/.vscode-server/extensions/ms-python.python-2020.5.80290/out/client/extension.js:92:841650)	at async d.updateTestSettings (/home/rvoigt/.vscode-server/extensions/ms-python.python-2020.5.80290/out/client/extension.js:92:841277)] {  code: 'FileNotFound',  name: 'EntryNotFound (FileSystemError)'}
```

That's my remote SSH box, which doesn't have VSCode installed on it except whatever node process the remote-SSH plugin installs. It looks like it's expecting to find a settings.json on the remote machine and isn't creating one automatically in the absence of one?

I've created the `rvoigt/.config/Code/User dir and an empty `settings.json` file in there; I'll update on whether or not it works better next time I run into this.

**Comment by RVoigtHearsay** [comment]
Update: Adding the empty settings.json file in the location where the error seems to be expecting it didn't help. It stops the error from showing in the developer tools, but the UX remains the same.

I also updated the settings.json in both locations with the correctly-formatted `remote.SSH.remotePlatform` entry, sadly to no avail there either.

**Comment by roblourens** [comment]
Hey @sandy081 I am trying to update a setting in local User settings, it looks like it's trying to write it to remote user settings instead, any idea why? It's a `application` scoped setting, and I call update with `vscode.ConfigurationTarget.Global`

**Comment by sandy081** [comment]
No idea, unless there is a bug. I can help you in debugging if you can repro.

**Comment by vilic** [comment]
Any update on this? Is this a bug of remote-ssh or vscode? It was working in our case.

**Comment by roblourens** [comment]
I can't repro it and I'm not sure what the issue is. 

If you set the setting manually in your User settings, in the format I described here, it should work: https://github.com/microsoft/vscode-remote-release/issues/2934#issuecomment-641479109 Please let me know if it still isn't working after adding that setting.

**Comment by RVoigtHearsay** [comment]
I've discovered a contributing factor: If the `settings.json` file doesn't validate perfectly (i.e.; if there are trailing commas or commented-out lines) then the VSCode settings editor UI will produce an error when trying to modify settings. I don't think that used to be the case (I could swear it was less strict about JSON validation and allowed comments and trailing commas in the past).

I noticed I was getting this error when trying to configure some things totally unrelated to vscode-remote; and when I fixed my settings.json so that it fully passes JSON validation, it also fixed the problem of it forgetting the remote machine type for vscode-remote.

However, all that said, my own complaint about this is as much about the poor UX as about the functionality. Even if the settings.json is invalid, throwing modal, interruptive dialogs at the user on a window reload is still a really unpleasant experience.

**TL;DR: Validate settings.json at all scopes (user, workspace, remote, etc.) and ensure that they contain only strictly valid JSON.**

**Comment by roblourens** [comment]
Oh, that's interesting. There should be no problem with comments or trailing commas in settings.json. Are you sure that's the problem and there isn't some other syntax issue?

**Comment by vilic** [comment]
Interestingly, the error seems to be gone after I reinstall the SSH extension.

**Comment by vilic** [comment]
@roblourens Hi, this is still an issue for us:

```
[00:18:50.446] Unhandled rejection: Error: Unable to write to User Settings because remote.SSH.remotePlatform is not a registered configuration.
    at D.reject (file:///C:/Users/vilicvane/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5545:245)
    at D.resolveAndValidate (file:///C:/Users/vilicvane/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5548:116)
    at D.doWriteConfiguration (file:///C:/Users/vilicvane/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5542:414)
    at Object.factory (file:///C:/Users/vilicvane/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5542:190)
    at u.consume (file:///C:/Users/vilicvane/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:130:421)
    at file:///C:/Users/vilicvane/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:130:236
    at new Promise (<anonymous>)
    at u.queue (file:///C:/Users/vilicvane/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:130:160)
    at D.writeConfiguration (file:///C:/Users/vilicvane/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5542:175)
    at C.writeConfigurationValue (file:///C:/Users/vilicvane/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5570:356)
    at file:///C:/Users/vilicvane/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5558:594
    at processTicksAndRejections (internal/process/task_queues.js:85:5)
```

Sometimes if I close all VSCode instances, and reopen the workspace, it works. But sometimes it doesn't.
